crates/bevy_core_pipeline/src/tonemapping/node.rs

```diff
@@ -1,22 +1,36 @@
 use std::sync::Mutex;
 
-use crate::tonemapping::{TonemappingPipeline, ViewTonemappingPipeline};
+use crate::tonemapping::{TonemappingLuts, TonemappingPipeline, ViewTonemappingPipeline};
+
 use bevy_ecs::prelude::*;
 use bevy_ecs::query::QueryState;
 use bevy_render::{
+    render_asset::RenderAssets,
     render_graph::{Node, NodeRunError, RenderGraphContext, SlotInfo, SlotType},
     render_resource::{
         BindGroup, BindGroupDescriptor, BindGroupEntry, BindingResource, LoadOp, Operations,
         PipelineCache, RenderPassColorAttachment, RenderPassDescriptor, SamplerDescriptor,
         TextureViewId,
     },
     renderer::RenderContext,
-    view::{ExtractedView, ViewTarget},
+    texture::Image,
+    view::{ExtractedView, ViewTarget, ViewUniformOffset, ViewUniforms},
 };
 
+use super::{get_lut_bindings, Tonemapping};
+
 pub struct TonemappingNode {
-    query: QueryState<(&'static ViewTarget, &'static ViewTonemappingPipeline), With<ExtractedView>>,
+    query: QueryState<
+        (
+            &'static ViewUniformOffset,
+            &'static ViewTarget,
+            &'static ViewTonemappingPipeline,
+            &'static Tonemapping,
+        ),
+        With<ExtractedView>,
+    >,
     cached_texture_bind_group: Mutex<Option<(TextureViewId, BindGroup)>>,
+    last_tonemapping: Mutex<Option<Tonemapping>>, // TODO Griffin use change detection?
```

Comment:
You could do change detection on the main world component during extraction and mark in the extracted component when it has changed so that this code can use that to know when it needs to create a new binding.

Reply:
Since the extracted component is now just an enum I think it would be better to keep this for now.